archiver.go

```diff
@@ -228,12 +231,19 @@ func (a *Archiver) createSymlink(path string, fi os.FileInfo, hdr *zip.FileHeade
 	a.m.Lock()
 	defer a.m.Unlock()
 
-	w, err := a.createHeader(fi, hdr)
+	link, err := os.Readlink(path)
 	if err != nil {
 		return err
 	}
 
-	link, err := os.Readlink(path)
+	// Don't use a data descriptor to shave a few bytes and to make sure that the symlink can be stream-unzipped
+	hdr.Flags &= ^uint16(0x8)
+	hdr.Method = zip.Store
+	hdr.CompressedSize64 = uint64(len(link))
+	hdr.UncompressedSize64 = hdr.CompressedSize64
+	hdr.CRC32 = crc32.Update(0, crcTable, []byte(link))
```

Comment:
@michalc Maybe we can just use https://pkg.go.dev/hash/crc32#ChecksumIEEE here?

Reply:
Ah @saracen I think so! Now changed